x-pack/plugins/maps/public/layers/vector_layer.js

```diff
@@ -175,8 +175,7 @@ export class VectorLayer extends AbstractLayer {
   }
 
   async getBounds(dataFilters) {
-    const isStaticLayer =
-      !this.getSource().isBoundsAware() || !this.getSource().isFilterByMapBounds();
+    const isStaticLayer = !this.getSource().isBoundsAware();
```

Comment:
removed isFilterByMapBounds check. This resulted in weird behavior where Elasticsearch layers would not call geo_bounds unless "dynamic filter" was selected